testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/authentication/policy/AuthenticationPolicyContextTestCase.java

```diff
@@ -0,0 +1,323 @@
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.ws.api.configuration.ClientConfigUtil;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.picketlink.common.exceptions.fed.WSTrustException;
+import org.picketlink.identity.federation.api.wstrust.WSTrustClient;
+import org.picketlink.identity.federation.core.saml.v2.util.DocumentUtil;
+import org.picketlink.identity.federation.core.wstrust.plugins.saml.SAMLUtil;
+import org.picketlink.trust.jbossws.SAML2Constants;
+import org.w3c.dom.Element;
+
+import javax.xml.namespace.QName;
+import javax.xml.ws.BindingProvider;
+import javax.xml.ws.Service;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.InetAddress;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.fail;
+
+/**
+ * Before the fix web service with STS (Picketlink) returned wrong subject with PolicyContext("javax.security.auth.subject.container").
+ * This test calls SimpleSecurityManager and checks if no exception is thrown and PolicyContext.getContext("javax.security.auth.Subject.container") should not be null.
+ * Test for [ WFLY-8946 ].
+ *
+ * @author Daniel Cihak
+ */
+@RunWith(Arquillian.class)
+public class AuthenticationPolicyContextTestCase {
+
+    private static Logger LOGGER = Logger.getLogger(AuthenticationPolicyContextTestCase.class);
+
+    private static final String PICKETLINK_STS = "picketlink-sts";
+    private static final String PICKETLINK_STS_WS = "picketlink-sts-ws";
+    private static final String RESOURCE_DIR = "org/jboss/as/test/integration/ws/authentication/policy/resources/";
+
+    private static final String HOST = TestSuiteEnvironment.getServerAddress();
+    private static final int PORT_OFFSET = 0;
+    private static final String USERNAME = "UserA";
+    private static final String PASSWORD = "PassA";
+    private static final String DEFAULT_HOST = getHostname();
+    private static final int DEFAULT_PORT = 8080;
+
+    private volatile ModelControllerClient modelControllerClient;
+    private static WSTrustClient wsClient;
+    private volatile CommandContext commandCtx;
+    private volatile ByteArrayOutputStream consoleOut = new ByteArrayOutputStream();
+
+    @ArquillianResource
+    private static volatile Deployer deployer;
+
+    @Deployment(name = PICKETLINK_STS, managed = false, testable = false)
+    public static Archive<?> createPicketlinkStsDeployment() {
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
```

Comment:
You should not need the TCCL, arquillian already has methods that add resources from the current class loader, have a look at how other tests do it.